### ISSUE TITLE
[Bug]: KafkaIO Batch Write Failing due to producer cannot commit messages within timeout

### ISSUE THREAD

**Issue by Abacn** [apache/beam#24963]
### What happened?

Found by Kafka performance test failing after #24879 in. This is because we removed shuffle=appliance there and then read from shuffle becomes faster. However, the producer is unable to digests data within timeout (can be seen by the flooding warning log of `send failed : 'Expiring 148 record(s) for beam-sdf-0:130675 ms has passed since batch creation'`). However, the message itself does not contain a timestamp and should be tolerant to throttling.

The error happens because producer.send is asynchronous. It adds a system timestamp when .send gets called and returns immediately. We need a mechanism to prevent overwhelming Kafka producer.

### Issue Priority

Priority: 2 (default / most bugs should be filed as P2)

### Issue Components

- [ ] Component: Python SDK
- [X] Component: Java SDK
- [ ] Component: Go SDK
- [ ] Component: Typescript SDK
- [X] Component: IO connector
- [ ] Component: Beam examples
- [ ] Component: Beam playground
- [ ] Component: Beam katas
- [ ] Component: Website
- [ ] Component: Spark Runner
- [ ] Component: Flink Runner
- [ ] Component: Samza Runner
- [ ] Component: Twister2 Runner
- [ ] Component: Hazelcast Jet Runner
- [ ] Component: Google Cloud Dataflow Runner

**Comment by aromanenko-dev** [comment]
Kind ping. What is status of this issue?

**Comment by aromanenko-dev** [comment]
@Abacn Yes, throttling detection is a general problem across all IO connectors and runner. So, should we close this one then?
